pkg/encrypt/encrypt.go

```diff
@@ -0,0 +1,113 @@
+package encrypt
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"bytes"
+	"context"
+	"crypto/cipher"
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"strings"
+
+	"golang.org/x/crypto/chacha20poly1305"
+
+	"github.com/Azure/ARO-RP/pkg/env"
+)
+
+const prefix = "ENC*"
+
+type Cipher interface {
+	Decrypt([]byte) ([]byte, error)
+	Encrypt([]byte) ([]byte, error)
```

Comment:
I wonder whether it will turn out to be better for Encrypt to return string, error; and Decrypt to take in a string.

Reply:
I found []byte easier to use in the code. Less casting and conversions. I think `[]byte` are easier

Reply:
I suspect that using []byte instead of string might be what is responsible for you needing RawBytesExt.